python/paddle/distributed/auto_parallel/api.py

```diff
@@ -55,15 +58,46 @@ def __init__(self, mesh, sharding_specs):
             for dim_name in sharding_specs
         ), 'The dimension name in sharding_specs must be an instance of str.'
 
+        self._sharding_specs = sharding_specs
         dims_mapping = [
             mesh.dim_names.index(dim_name) if dim_name is not None else -1
             for dim_name in sharding_specs
         ]
 
         # 2. init core.TensorDistAttr
         core.TensorDistAttr.__init__(self)
-        self.process_mesh = mesh
-        self.dims_mapping = dims_mapping
+        self._process_mesh = mesh
```

Comment:
the process_mesh field in c++ TensorDistAttr might remain unchanged ?

Reply:
fixed